### ISSUE TITLE
Cannot read property 'fetchingFiles' of undefined

### ISSUE THREAD

**Issue by vjpr** [pnpm/pnpm#796]
```
node-sass 4.5.3                                           linking 4.5.3 ↓
  ERROR Cannot read property 'fetchingFiles' of undefined
```

```
{"time":1497529896796,"hostname":"Vaughan-MacBook-Pro.local","pid":41712,"level":"error","name":"pnpm","message":{},"err":{"name":"TypeError","message":"Cannot read property 'fetchingFiles' of undefined","stack":"TypeError: Cannot read property 'fetchingFiles' of undefined\n    at /Users/Vaughan/nvm/versions/node/v8.1.1/lib/node_modules/pnpm/lib/link/index.js:127:46\n    at Generator.next (<anonymous>)\n    at /Users/Vaughan/nvm/versions/node/v8.1.1/lib/node_modules/pnpm/lib/link/index.js:7:71\n    at Promise (<anonymous>)\n    at __awaiter (/Users/Vaughan/nvm/versions/node/v8.1.1/lib/node_modules/pnpm/lib/link/index.js:3:12)\n    at linkPkg (/Users/Vaughan/nvm/versions/node/v8.1.1/lib/node_modules/pnpm/lib/link/index.js:126:12)\n    at limitLinking (/Users/Vaughan/nvm/versions/node/v8.1.1/lib/node_modules/pnpm/lib/link/index.js:117:66)\n    at run (/Users/Vaughan/nvm/versions/node/v8.1.1/lib/node_modules/pnpm/lib/node_modules/p-limit/index.js:22:4)\n    at Promise (/Users/Vaughan/nvm/versions/node/v8.1.1/lib/node_modules/pnpm/lib/node_modules/p-limit/index.js:35:4)\n    at Promise (<anonymous>)"}}
```

### Additional information:

 - `node -v` prints: 8.1.1
 - Windows, OS X, or Linux?: macOS

**Comment by vjpr** [comment]
Seems to only happen when using `--force`.

**Comment by zkochan** [comment]
I cannot reproduce for some reason. Is it the only dependency in the package or does it happen with some specific set of dependencies?

**Comment by vjpr** [comment]
It happens when running a `pnpm link`. Somehow `dependency` is null. Should be able to find the issue from static types...

**Comment by zkochan** [comment]
This is happening because argumentless forced installation fails... I will publish a fix shortly

**Comment by vjpr** [comment]
Still occurs in 0.69.6.

**Comment by zkochan** [comment]
This is the issue as #870, fixed in [pnpm@1.14.7](https://github.com/pnpm/pnpm/releases/tag/v1.14.7) via https://github.com/pnpm/supi/commit/bc800a86920f743ea2d7dd70dbe3814ab2f6e873
